hera_cal/abscal.py

```diff
@@ -3336,6 +3336,72 @@ def build_data_wgts(data_flags, data_nsamples, model_flags, autocorrs, auto_flag
     return DataContainer(wgts)
 
 
+def _get_idealized_antpos(cal_flags, antpos, pols, tol=1.0, keep_flagged_ants=True, data_wgts={}):
+    '''Figure out a set of idealized antenna positions that doesn't introduce additional 
+    redcal degeneracies.
+    
+    Arguments:
+        cal_flags: dictionary mapping keys like (1, 'Jnn') to flag waterfalls
+        antpos: dictionary mapping antenna numbers to numpy array positions
+        pols: list of polarizations like ['ee', 'nn']
+        tol: float distance for baseline match tolerance in units of baseline vectors (e.g. meters)
+        keep_flagged_ants: If True, flagged antennas that are off-grid (i.e. would introduce an 
+            additional degeneracy) are placed at the origin. Otherwise, flagged antennas in cal_flags
+            are excluded from idealized_antpos.
+        data_wgts: DataContainer mapping baselines like (0, 1, 'ee') to weights. Used to check if 
+            flagged antennas off the calibratable grid have no weight. Ignored if keep_flagged_ants
+            is False.
+
+    Returns:
+        idealized_antpos: dictionary mapping antenna numbers to antenna positions on a perfect grid, 
+            but arbitarily skewed from the input antpos.
+    '''
+    # build list of reds without flagged untennas
+    all_ants = list(cal_flags.keys())
+    unflagged_ants = [ant for ant in cal_flags if not np.all(cal_flags[ant])]
+    all_reds = redcal.get_reds(antpos, bl_error_tol=tol, pols=pols)
+    unflagged_reds = redcal.filter_reds(all_reds, ants=unflagged_ants)
+    
+    # count the number of dimensions describing the redundancies of unflagged antennas
+    unflagged_idealized_antpos = redcal.reds_to_antpos(unflagged_reds, tol=redcal.IDEALIZED_BL_TOL)
+    unflagged_nDims = _count_nDims(unflagged_idealized_antpos, assume_2D=False)
+    
+    # get the potentially calibratable ants, reds, and idealized_antpos. These are antennas may be
```

Comment:
Thes are antennas may be -> These are antennas that may be

Reply:
Fixed.